.codex/hooks/stop_retry.py

```
@@ -0,0 +1,164 @@
+def save_state(state):
+    STATE_PATH.parent.mkdir(parents=True, exist_ok=True)
+    STATE_PATH.write_text(json.dumps(state, indent=2))
+
+
+def reset_state():
+    save_state({"count": 0, "last_counted_turn_id": None})
+
+
+def get_staged_files():
+    proc = subprocess.run(
+        ["git", "diff", "--cached", "--name-only"],
```

Ruff (0.15.12)

[error] Starting a process with a partial executable path

(S607)